### PR TITLE
RestClient Module-Level Logging

### PR DESCRIPTION
**Pull request by ric-evans**
Now, `rest_tools.RestClient` uses a module-level logger, which allows clients to suppress/limit these logs.

closes https://github.com/WIPACrepo/rest-tools/issues/6

### CODE REVIEW

**Comment by ric-evans** [comment]
PyJWT apparently made some non-backwards compatible changes: https://pyjwt.readthedocs.io/en/2.0.1/changelog.html#jwt-encode-return-type

**Comment by ric-evans** [comment]
PyJWT also changed `decode()`'s required params: https://pyjwt.readthedocs.io/en/2.0.1/changelog.html#require-explicit-algorithms-in-jwt-decode-by-default

File: rest_tools/server/auth.py [hunk 1]
```diff
@@ -46,7 +50,12 @@ def create_token(self, subject, expiration=None, type='temp', payload=None):
             'iat': now,
             'type': type,
         })
-        return jwt.encode(payload, self.secret, algorithm=self.algorithm).decode('utf-8')
+
+        token = jwt.encode(payload, self.secret, algorithm=self.algorithm)
+        try:
+            return token.decode("utf-8")
+        except (UnicodeDecodeError, AttributeError):
+            return token
```
Comment (dsschult) [inline rest_tools/server/auth.py:58]: We can probably just drop `.decode('utf-8')` and be done with it, unless you think we should maintain compatibility with older versions of pyjwt.

**Review by dsschult — APPROVED** [review]

**Comment by dsschult** [comment]
Oh, I guess the one thing you haven't done is change the requirements.txt file to require >= 2.0

File: rest_tools/server/auth.py (hunk 1 above)
Comment (ric-evans) [reply rest_tools/server/auth.py:58]: I'm fine either way. Not maintaining backwards compatibility could break scripts that are using rest-tools _and_ <2.0 pyjwt. I'm not sure if that's something we should be aware of

Comment (dsschult) [reply rest_tools/server/auth.py:58]: I think most things are only using pyjwt through rest-tools.  There might be one or two special projects, but they shouldn't be special and indicates code should be upstreamed to rest-tools.

Comment (dsschult) [reply rest_tools/server/auth.py:58]: So, I'm fine with upgrading and breaking old things, as long as you update `requirements.txt`

**Pull request merged**